### PR TITLE
[111, 88] Lot profile (Seller/Buyer) (Frontend)

### PR DESCRIPTION
**Pull request by SaidganiUsm**
Where
[Browse auctions (Frontend)](https://jjbcapital.atlassian.net/browse/NL-94)
[View seller lot profile (Frontend)](https://jjbcapital.atlassian.net/browse/NL-88)

What
View for lot profile (seller and buyer)

How
View lot profile for buyer and seller with routing and all functions

Affected Area
Auctionify.API

PR Checklist
Build

 Build Succeeded
Unit Tests

 X Tests Passed, Y Ignored, Z Failed

### CODE REVIEW

File: src/Auctionify.API/ClientApp/src/app/web-api-client.ts [hunk 1]
```diff
@@ -267,6 +267,28 @@ export class Client {
         );
     }
 
+    getOneLotForBuyer(id: number): Observable<BuyerGetLotResponse> {
+        let url_ = this.baseUrl + `/api/lots/${id}/buyer`;
```
Comment (khasanrashidov) [inline src/Auctionify.API/ClientApp/src/app/web-api-client.ts:271]: Please fix it, it is, '/buyers' (in plural):

let url_ = this.baseUrl + `/api/lots/${id}/buyers`;

**Review by khasanrashidov — COMMENTED** [review]
and please merge with dev as well, resolve conflicts

**Comment by khasanrashidov** [comment]
1. Your branch is outdated and doesn't have SignalR (which is in dev), I suggest upgrading the your branch and integrating it.

2.  why don't we keep consistency and call the route (path) to get lot as 'get-lot/:id'

image

3. I could not get the lot profile for seller: (used https://localhost:44452/seller/lot-profile/2)

image

File: src/Auctionify.API/ClientApp/src/app/components/buyer/buyer-routing.module.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        path: '',
+        redirectTo: 'home',
+        pathMatch: 'full',
+    },
+    {
+        path: 'lot-profile/:id',
+        component: LotProfileBuyerComponent,
+        pathMatch: 'full',
+    }
```
Comment (otash8110) [inline src/Auctionify.API/ClientApp/src/app/components/buyer/buyer-routing.module.ts:15]: I suggest making lot-profile general path, not only for buyer or seller. So, please move it to general routing module

**Review by otash8110 — CHANGES_REQUESTED** [review]
image

- Add styles for the block of images on the left from main image. 
- Add blue block from design which contains price information on right side
- Merge dev with bids functionality and implement bids block on right side
- I cannot get a lot profile as a seller

image
-Fix images block for cases when there are no any images

image
- If there is a parect category it shows id, not parent category name

image
- fix documents and files styles

**Comment by khasanrashidov** [comment]
Here can you show the actual number of bids for the lot, as it is in figma (we have it in the response of API as bidCount):

image_2023-12-07_11-30-25

what we have in figma:

image

I don't know if I am going too much on that (I mean styling), but these styles differ, and does it matter (for below)?
### what we have:
image
### what is in figma design:
image

**Comment by khasanrashidov** [comment]
is it possible to make the image clickable, so that when I click the image it shows the larger one, right now it is not possible (or if it is not, then you may consider not to include it:

image

**Review by khasanrashidov — APPROVED** [review]

**Review by khasanrashidov — APPROVED** [review]
Checked, works good! Requested changes were applied, looks good to me.

update:
fix issue below

File: src/Auctionify.API/ClientApp/src/app/components/general/lot-profile/lot-profile.component.html [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+                        <div class="right__bid-number-date">
+                            <div class="right__bid-number">
+                                <p>{{ lotData.bids.length - i }}</p>
+                            </div>
+                            <div class="right__bid-date">
+                                <p>{{ formatBidDate(bid.timeStamp) }}</p>
+                            </div>
+                        </div>
+                        <div class="right__bid-price">
+                            <p>{{ bid.newPrice | currency: 'USD' }}</p>
```
[the remark this answers is not in the document]
Comment (khasanrashidov) [reply src/Auctionify.API/ClientApp/src/app/components/general/lot-profile/lot-profile.component.html]: Fixed thank you!

**Review by khasanrashidov — APPROVED** [review]

**Review by otash8110 — CHANGES_REQUESTED** [review]
image
When I try to change user data I get validation error without message

image
- Log out is not working because of the wrong type of the dialog (Should be just Dialog) and overload method of the open. See other components which have such functionality.

image
- When I click on eye to show hidden password the form is submited. Also add some space between inputs because message cannot fit

**Review by otash8110 — DISMISSED** [review]

**Review by Muhammad-gf — APPROVED** [review]

**Review by otash8110 — APPROVED** [review]

**Review by khasanrashidov — APPROVED** [review]

**Pull request merged**